### PR TITLE
update Flask examples

### PR DESCRIPTION
**Pull request by bfineran**
Introduce `engine_flask_server()` and `EngineFlaskClient` that can be used as simple client/server wrapper for the `Engine` class.

### CODE REVIEW

**Review by mgoin — DISMISSED** [review]
LGTM, thanks!

File: examples/flask/server.py [hunk 1]
```diff
@@ -48,10 +48,66 @@
...
+    :param num_sockets: The number of physical sockets to run the model on.
+        Pass None or 0 to run on the max number of sockets for the
+        current machine, default None
+    :param scheduler: The kind of scheduler to execute with. Pass None for the default.
+    :param address: IP address to run on. Default is 0.0.0.0
+    :param port: port to run on. Default is 5543
+    :return: launches a flask server on the given address and port can run the
+        given model on the DeepSparse engine via HTTP requests
+    """
+    print(f"Compiling model at {model_path}")
```
Comment (kevinaer) [inline examples/flask/server.py:82]: Could we potentially change the print statements to logs? Specially for the flask API endpoints.

Comment (markurtz) [reply examples/flask/server.py:82]: agree with Kevin, would be good to get these moved to logs now and potentially add logging into the classes

**Review by mgoin — DISMISSED** [review]

File: examples/flask/server.py [hunk 2]
```diff
@@ -44,12 +52,72 @@
...
+
+_LOGGER = log_init(os.path.basename(__file__))
+
+
+def engine_flask_server(
+    model_path: str,
+    batch_size: int = 1,
+    num_cores: int = None,
+    num_sockets: int = None,
+    scheduler: Scheduler = None,
```
Comment (bnellnm) [inline examples/flask/server.py:72]: Shouldn't this default to the multi-stream scheduler?

Comment (bfineran) [reply examples/flask/server.py:72]: I believe this should be handled by the `compile_model` function.
Was going off of the defaults [here](https://github.com/neuralmagic/deepsparse/blob/main/src/deepsparse/engine.py#L548).

Comment (bfineran) [reply examples/flask/server.py:72]: are you saying that for a client/server scenario, we want to instead default to multi-stream?

Comment (bfineran) [reply examples/flask/server.py:72]: that makes sense to me, so going with that.

Comment (bnellnm) [reply examples/flask/server.py]: Yeah, that was the general idea for multi-stream.  Unless @mgoin or anyone else has objections.  It should be fine either way for an example.

Comment (bfineran) [reply examples/flask/server.py]: makes sense. thanks for the catch @bnellnm

**Review by markurtz — APPROVED** [review]

**Review by mgoin — APPROVED** [review]

**Pull request merged**